### ISSUE TITLE
semgrep v1.94.0+ fails to execute when in a directory outside of PATH

### ISSUE THREAD

**Issue by lae** [semgrep/semgrep#10652]
**Describe the bug**

Executing `path/to/semgrep` installed in a Python venv fails to run since v1.94.0, with the following error:

```
[00.04][ERROR]: Error: exception Unix_error: No such file or directory execvp pysemgrep
Raised by primitive operation at CLI.safe_run in file "src/osemgrep/cli/CLI.ml", line 272, characters 8-12
```

9eee60510cdbfdb4c382c8945ed604bd978b496c moved around files a bit which seems to have broken the expectations (?) of the logic in https://github.com/semgrep/semgrep/blob/9eee60510cdbfdb4c382c8945ed604bd978b496c/cli/src/semgrep/console_scripts/entrypoint.py#L43-L53

and is no longer able to find the `pysemgrep` script because `$VENV/bin/` isn't being added to the PATH. 

**To Reproduce**

1. `python3 -m venv /semgrep` (create venv)
2. `/semgrep/bin/pip install semgrep==1.94.0` (install stable semgrep, earliest affected version)
3. `/semgrep/bin/semgrep --version` (execute semgrep)

**Expected behavior**

`semgrep` should still launch and execute as it did in 1.93.0 and before.

**What is the priority of the bug to you?**

- [ ] P0: blocking your adoption of Semgrep or workflow
- [ ] P1: important to fix or quite annoying
- [x] P2: regular bug that should get fixed

Setting the PATH on launch resolves it (see https://github.com/amplify-security/runner/pull/20, which provides the context for this issue) , but I think this was unintended breakage so it should probably be fixed.

**Comment by cgdolan** [comment]
Thanks so much for reporting this issue and for the detailed reproduction steps! Would you mind having a look at https://github.com/semgrep/semgrep/pull/10658 and letting me know if this fix seems reasonable?

**Comment by lae** [comment]
That does seem like a reasonable fix, yeah.

**Comment by cgdolan** [comment]
The fix has been merged and will be included in our next release. I'm going to keep this issue open until the next version is published to pypi and I've verified the fix fully end-to-end.

**Comment by aryx** [comment]
FYI I don't think I've closed the issue; it just got closed automatically when your PR was merged I think.
(or maybe I did close it ... then sorry about that).

**Comment by cgdolan** [comment]
Yeah, no worries. It auto-closed because I tagged this issue in the PR description.

**Comment by cgdolan** [comment]
Closing now that the fix is released in v1.96.0 and verified with:

```
chris@chriss-mbp ~ % docker run --rm -it python:3.12 /bin/bash
root@5cf44c158259:/# python3 -m venv /semgrep
root@5cf44c158259:/# /semgrep/bin/pip install semgrep==1.96.0
root@5cf44c158259:/# /semgrep/bin/semgrep --version
1.96.0
```
